source/shared/cpp/ObjectModel/ActionParserRegistration.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "pch.h"
+#include "Enums.h"
+#include "json\json.h"
+
+namespace AdaptiveCards
+{
+    class BaseActionElement;
+    class ElementParserRegistration;
+    class ActionParserRegistration;
+
+    class IActionElementParser
+    {
+    public:
+        virtual std::shared_ptr<BaseActionElement> Deserialize(
```

Comment:
>Deserialize [](start = 51, length = 11)

Discussed offline. Let's try to also add a version of these that don't require you to pass the registrations

Reply:
Discussed offline some more.  This interface defines the element parsers.  We don't want custom parsers to need two versions, and the only people likely to need our element parsers are those implementing custom parsers (who will have the registration handy).  The top level deserialize method takes the registration as an optional parameter.  Leaving these as is.

---
In reply to: [146913501](https://github.com/Microsoft/AdaptiveCards/pull/810#discussion_r146913501) [](ancestors = 146913501)